src/meson.build

```diff
@@ -0,0 +1,54 @@
+message('----- CPP -----')
+cpp = meson.get_compiler('cpp')
+message(cpp.get_id())
+message(cpp.version())
+message(meson.project_source_root())
+message(py3.get_install_dir())
+message(py3.language_version())
+message('---------------')
+
+incdir_pybind11 = run_command(py3,
+  ['-c', 'import pybind11; print(pybind11.get_include())'],
+  check: true
+).stdout().strip()
+
+inc_pybind11 = include_directories(incdir_pybind11)
+
+pybind11_config = find_program('pybind11-config')
+pybind11_config_ret = run_command(pybind11_config, ['--includes'], check: true)
+pybind11_dep = declare_dependency(
+  include_directories: [pybind11_config_ret.stdout().split('-I')[-1].strip()],
+)
+
+python3_dep = dependency('python3', )
```

Comment:
Since Meson 0.63.0, you don't need to explicitly pass the python dependency to `.extension_module()`, but if you do pass it, it should be the one from `py3.dependency()` as that is guaranteed to be the same version of python that you're building modules for. `dependency('python3')` may pick up a different system dependency via pkg-config, if there are multiple versions of python installed.

Reply:
Thanks @eli-schwartz, I am very happy to receive help from `meson` experts!

Reply:
No problem. :) BTW you can also prune the `python3_dep` variable I think, since it is not actually used now.